popup.js

```diff
@@ -243,22 +259,27 @@ const saveWatchlistToLocal = (multimedias) => {
   );
 };
 
+const randomizeFromCacheBtn = document.querySelector("#randomizeFromSync");
+randomizeFromCacheBtn.hasEventListener = false;
 const loadWatchlistFromLocal = (randomizeFromCacheBtn, listenerCBack) => {
```

Comment:
The problem in this function was that every time u click the "scan & rand" button a new event listener for the "Randomize" Button is added, so for example if u clicked "scan & rand" button 3 times and after that u clicked "Randomize" button only once, it would be considered as it was clicked 3 times, because there is 3  event listener created.
Here, I solved that problem by using event listener flag.

Reply:
Great catch!
But I feel like our approach is wrong on solving this. What you did is like the cleanup function in a `React.useEffect` whereas here we have full control on plain JS. I think a better question would be why are letting users click on it multiple times in the first place? I was really amateur when I first made this extension, so looks like I couldn't cover UX very nice.

I think we can just block users to click on it several times or we can just restart the process from zero once clicked again, but I don't see a use case for the latter.

Reply:
If we block the button, users will have no way to update their saved watchlist.

Reply:
We should block the click while the scan is in progress and enable again after.

Reply:
You didn't get me well, the problem happen when someone click scan & randomize button even if scan is done. 
here's a video to make it more clear.   


https://github.com/AlkTheOrg/imdb-watchlist-randomizer/assets/110132520/1d1abbaf-7cda-49e6-b1f1-59e218a8b462

Reply:
Then this is an extremely rare case, because if you close the popup js all event listeners will be lost right? Although I have some feedback to give you on your code, I don't think we need that logic at all. Also the main problem is my initial solution which is using the multimedia from the callback's stale data which is a very bad practice in this case, but as I said I was very amateur when I made this extension.

Reply:
ok, u are right here.